vcsclient/bitbucketcloud.go

```diff
@@ -471,6 +473,66 @@ func (client *BitbucketCloudClient) GetRepositoryEnvironmentInfo(ctx context.Con
 	return RepositoryEnvironmentInfo{}, errBitbucketGetRepoEnvironmentInfoNotSupported
 }
 
+func (client *BitbucketCloudClient) GetModifiedFiles(ctx context.Context, owner, repository, refBefore, refAfter string) ([]string, error) {
+	err := validateParametersNotBlank(map[string]string{
+		"owner":      owner,
+		"repository": repository,
+		"refBefore":  refBefore,
+		"refAfter":   refAfter,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	bitbucketClient := client.buildBitbucketCloudClient(ctx)
+	options := &bitbucket.DiffStatOptions{
+		Owner:    owner,
+		RepoSlug: repository,
+		// We use 2 dots for spec because of the case described at the page:
+		// https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/#two-commit-spec
+		// As there is no `topic` set it will be treated as `refBefore...refAfter` actually.
+		Spec:    refBefore + ".." + refAfter,
+		Renames: true,
+		Merge:   true,
+	}
+
+	var fileNamesList []string
+	fileNamesSet := datastructures.MakeSet[string]()
+	nextPage := 1
+
+	for nextPage > 0 {
+		options.PageNum = nextPage
+		diffStatRes, err := bitbucketClient.Repositories.Diff.GetDiffStat(options)
+		if err != nil {
+			return nil, err
+		}
+
+		if diffStatRes.Next == "" {
+			nextPage = -1
+		} else {
+			nextPage++
+		}
+
+		addMissing := func(src map[string]any) {
+			if path, found := src["path"]; found {
+				if fileName, ok := path.(string); ok {
+					if !fileNamesSet.Exists(fileName) {
+						fileNamesList = append(fileNamesList, fileName)
+					}
+					fileNamesSet.Add(fileName)
+				}
+			}
+		}
```

Comment:
Can we put it in a new function outside of GetModifiedFiles?

Reply:
I have refactored it a bit, so now it is declared out of the loop and I removed `fileNamesList = append(fileNamesList, ...` as not needed.